x-pack/plugins/ingest_pipelines/public/application/components/pipeline_editor/components/processor_form/processor_form.container.tsx

```diff
@@ -33,6 +39,14 @@ interface Props {
   processor?: ProcessorInternal;
 }
 
+const formOptions: FormOptions = {
+  /**
+   * This is important for allowing configuration of empty text fields in certain processors that
+   * remove values from their inputs.
+   */
+  stripEmptyFields: false,
```

Comment:
This is quite an important change for this fix, it also means we need to add the appropriate serializer for all fields. I tested all forms locally and the LGTM. But it would be good to check them for any unexpected resulting `"foo": ""` entries.